oauth2/src/main/java/com/linecorp/armeria/client/auth/oauth2/OAuth2AuthorizationGrant.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.auth.oauth2;
+
+import java.util.concurrent.CompletionStage;
+
+import com.linecorp.armeria.common.auth.oauth2.AccessTokenCapsule;
+
+/**
+ * Represents an OAuth 2.0 Access Token Grant flow to obtain Access Token.
+ */
+@FunctionalInterface
+public interface OAuth2AuthorizationGrant {
```

Comment:
Question: do you think it is better to abbreviate `Authorization` into `Authz`?

Reply:
You like reducing the number of symbols used in the class name or you are concerned about some ambiguity? The user won't interact with those classes directly.
On the client, the user interacts with `OAuth2Client` and one of the available Grants `(OAuth2ClientCredentialsGrant` or `OAuth2ResourceOwnerPasswordCredentialsGrant`).
On the server, the user interacts with `OAuth2TokenIntrospectionAuthorizer`.

Reply:
Thanks for explanation. Sounds good as it is, then. :+1: